pkg/client/client.go

```diff
@@ -0,0 +1,39 @@
+package client
+
+import (
+	"context"
+	"log/slog"
+	"os"
+
+	camelv1 "github.com/apache/camel-k/pkg/apis/camel/v1"
+	camelclient "github.com/apache/camel-k/pkg/client"
+	"github.com/go-logr/logr/slogr"
+	controllerlog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var logger = slog.Default()
+
+type Client struct {
+	camelclient.Client
+}
+
+func NewClient() (*Client, error) {
+	controllerlog.SetLogger(slogr.NewLogr(logger.Handler()))
+
+	cl, err := camelclient.NewClient(false)
+	if err != nil {
+		logger.Error("failed to create k8s client for camel k", err)
+		os.Exit(1)
+	}
+
+	ip := &camelv1.IntegrationPlatformList{}
+	err = cl.List(context.Background(), ip)
+	if err != nil {
+		logger.Warn("Failed to find IntegrationPlatform.", err)
```

Comment:
I think you should return `nil, Error` here as well as the next condition

Reply:
You don't think we should be able to run it if the Integration Platform is not yet created?

Reply:
I mean is there a point? But I am not sure from the design perspective what is better, if stop the app or having it running and not working

Reply:
wonder if that should be a rediness check, not sure who will actually look at the logs tbh